src/misc/breeding.ts

```diff
@@ -0,0 +1,131 @@
+import { Material, Particle, Sound } from 'org.bukkit';
+import { Block } from 'org.bukkit.block';
+import { Levelled, Waterlogged } from 'org.bukkit.block.data';
+import { Chicken, Cow, EntityType, LivingEntity, Pig } from 'org.bukkit.entity';
+import { EntityBreedEvent } from 'org.bukkit.event.entity';
+import { ItemStack } from 'org.bukkit.inventory';
+
+const animalsWithCustomBreeding: EntityType[] = [
+  EntityType.COW,
+  EntityType.PIG,
+  EntityType.CHICKEN,
+];
+
+const BREED_WATER_LEVEL = 1; // Max 3
+const BREED_COMPOST_LEVEL = 4; // Max 8
+
+const SEED_BOWL = Material.DEAD_BRAIN_CORAL_FAN;
+const EMPTY_SEED_BOWL = Material.DEAD_TUBE_CORAL_FAN;
+
+function hasFoodSource(entity: LivingEntity): boolean {
+  const blocksNearEntity = blocksInRadiusOfEntity(entity, 2);
+
+  for (const block of blocksNearEntity) {
+    if (entity instanceof Chicken) {
+      if (block.type === SEED_BOWL) {
+        block.setType(EMPTY_SEED_BOWL);
+        const data = block.blockData;
+        if (data instanceof Waterlogged) data.setWaterlogged(false);
+        block.setBlockData(data);
```

Comment:
replace setters and getters ->

block.setType(...)   ->  block.type = ...

block.setBlockData(...)  --> block.blockdata = ...